src/optimizer/transforms/__tests__/char-class-classranges-merge-transform-test.js

```diff
@@ -100,6 +100,13 @@ describe('char-class-classranges-merge', () => {
     expect(re.toString()).toBe('/[\\ud83d\\ude80-\\ud83d\\ude88]/u');
   });
 
+  it('does not merge non \\w chars into class ranges to keep readability', () => {
```

Comment:
but why? optimization isn't about readability

Reply:
At the moment, the optimizer tries to find the right balance between optimization, strictness and readability. Once we add options, we'll be able to have more atomic settings.

Reply:
Yeah, I think it's a good trade-off, thanks guys!